core/src/main/java/com/oath/oak/NativeAllocator/OakNativeMemoryAllocator.java

```diff
@@ -142,43 +156,53 @@ public Slice allocateSlice(int size) {
     @Override
     public void free(ByteBuffer bb) {
         allocated.addAndGet(-(bb.remaining()));
-        if (stats != null) stats.release(bb);
-        freeList.add(
-            new AbstractMap.SimpleImmutableEntry<>(
-                freeCounter.getAndIncrement(), new Slice(INVALID_BLOCK_ID,bb)));
+        if (stats != null) {
```

Comment:
Do we still use free() of ByteBuffers?

Reply:
If this free/allocate are here just because of the interface, maybe to throw an unimplemented exception? Just as a sanity check...

Reply:
It is deprecated, but not deleted since it is used in some tests.